client/src/components/SelectionField/SelectionField.vue

```diff
@@ -0,0 +1,180 @@
+import { storeToRefs } from "pinia";
+import { computed, type Ref, ref, watch } from "vue";
+import Multiselect from "vue-multiselect";
+
+import {
+    getDataset,
+    getDatasetCollection,
+    getHistories,
+    getInvocations,
+    getJobs,
+    getWorkflows,
+} from "@/components/SelectionField/services";
+import { type EventData, useEventStore } from "@/stores/eventStore";
+import { useHistoryStore } from "@/stores/historyStore";
+
+import type { ApiResponse, OptionType } from "./types";
+
+import LoadingSpan from "@/components/LoadingSpan.vue";
+
+const eventStore = useEventStore();
+const { currentHistoryId } = storeToRefs(useHistoryStore());
+
+const DELAY = 300;
+
+const props = withDefaults(
+    defineProps<{
+        objectId?: string;
+        objectName?: string;
+        objectQuery?: (query: string) => Promise<ApiResponse>;
+        objectTitle?: string;
+        objectType: string;
+    }>(),
+    {
+        objectId: "",
+        objectName: "...",
+        objectQuery: undefined,
+        objectTitle: undefined,
+    }
+);
+
+const emit = defineEmits<{
+    (e: "change", newValue: OptionType): void;
+}>();
+
+const errorMessage = ref("");
+const dragData: Ref<EventData[]> = ref([]);
+const dragTarget: Ref<EventTarget | null> = ref(null);
+const dragState: Ref<"danger" | "success" | "warning" | null> = ref(null);
+const isLoading = ref(true);
+const options: Ref<Array<OptionType>> = ref([]);
+
+const currentValue = computed({
+    get: () => ({
+        id: props.objectId,
+        name: props.objectName,
+    }),
+    set: (newValue: OptionType) => {
+        emit("change", newValue);
+    },
+});
+
+// Drag-and-drop only when labels are not used
+const droppable = computed(() => ["history_dataset_id", "history_dataset_collection_id"].includes(props.objectType));
+
+const title = computed(
+    () =>
+        props.objectTitle || `Select a ${props.objectType.replace(/_/g, " ").replace(/\b\w/g, (c) => c.toUpperCase())}`
+);
+
+// Unified search behavior
+const search = debounce(async (query: string = "") => {
+    if (!errorMessage.value) {
+        try {
+            const data = await (props.objectQuery ? props.objectQuery(query) : doQuery(query));
+            isLoading.value = false;
+            errorMessage.value = "";
+            if (data) {
+                options.value = data.map((d: any) => ({ id: d.id, name: d.name ?? d.id, data: d }));
+            } else {
+                options.value = [];
+            }
+        } catch (e) {
+            errorMessage.value = String(e);
+        }
+    }
+}, DELAY);
+
+async function doQuery(query: string = ""): Promise<ApiResponse> {
+    switch (props.objectType) {
+        case "history_id":
+            return getHistories();
+        case "history_dataset_id":
+            return getDataset(query, currentHistoryId.value);
+        case "history_dataset_collection_id":
+            return getDatasetCollection(query, currentHistoryId.value);
```

Comment:
Related to the services changes:
```suggestion
            return currentHistoryId.value ? getDataset(query, currentHistoryId.value) : [];
        case "history_dataset_collection_id":
            return currentHistoryId.value ? getDatasetCollection(query, currentHistoryId.value) : [];
```